apps/initialize-markets/src/Markets.ts

```diff
@@ -0,0 +1,181 @@
+
+    // grab all accounts from graph protocol api, max limit is GRAPH_MAX_LIMIT per request
+    const accounts: Array<string> = [];
+    let tempAccounts: Array<string> = [];
+    let skip = 0;
+    do {
+      await wait(10);
+      tempAccounts = await fetch(graphUrl, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+
+        body: JSON.stringify({
+          query: `{
+          accounts(first: ${GRAPH_MAX_LIMIT}, skip: ${skip}, where:{
+              nextSettleTime_lte: "${(Date.now() / 1000).toFixed(0)}",
+              nextSettleTime_not: 0,
+          }${block ? `, block: {number: ${block}}` : ""}) {
+            id
+          }
+      }`,
+        }),
+      })
+        .then((r) => r.json())
+        // .then(r => {console.log(r); return r;})
+        .then((r: { data: { accounts: [{ id: string }] } }) =>
+          r.data.accounts.map((a) => a.id)
+        );
+      accounts.push(...tempAccounts);
+      skip += tempAccounts.length;
+    } while (tempAccounts.length === GRAPH_MAX_LIMIT);
+
+    console.log("accounts: ", accounts.length);
+    const perChunk = 100;
+    const accountsInChunks = accounts.reduce((resultArray, account, index) => {
+      const chunkIndex = Math.floor(index / perChunk)
+
+      if (!resultArray[chunkIndex]) {
+        resultArray[chunkIndex] = [] // start a new chunk
+      }
+
+      resultArray[chunkIndex].push(account)
+
+      return resultArray
+    }, [] as Array<Array<string>>)
+
+    if (accounts.length) {
+      return accountsInChunks.map((accountsChunk) => {
+        return {
+          to: settleAccountsAddress,
+          data: settleAccounts.encodeFunctionData("settleAccounts", [
+            accountsChunk,
+          ]),
+        };
+      });
+    } else {
+      console.log("No accounts to settle");
+      return [];
+    }
+  };
+
+  async getVaultAccountsSettlementTxs(block: number | null = null) {
+    const settleAccountsAddress = settleAccountsAddressMap[this.network];
+
+    const settleAccounts = new ethers.Interface(SETTLE_ACCOUNTS_ABI);
+
+    const vaultAccountsArray = await fetch(graphUrl, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+
+      body: JSON.stringify({
+        query: `{
+          balances(
+            where: {
+              account_: {systemAccountType: "None"},
+              token_: {tokenType: "VaultDebt", maturity_lt: ${(Date.now() / 1000).toFixed(0)}},
+              current_: {currentBalance_gt: 0}
+              }
+          ${block ? `, block: {number: ${block}}` : ""}) {
+            account {
+              id
+            }
+            token {
+              vaultAddress
+              maturity
+            }
+          }
+        }`
+      }),
+    })
+      .then((r) => r.json())
+      // group accounts per vault
+      .then((r: { data: { balances: [{ account: { id: string }, token: { vaultAddress: string } }] } }) =>
```

Comment:
there is a helper function `groupArrayToMap` or `groupArrayByKey` that you could use here.